pkg/webhook/deploy/deployment.go

```diff
@@ -46,6 +53,51 @@ func Webhook(ctx context.Context, cl runtimeclient.Client, s *runtime.Scheme, na
 	return nil
 }
 
+// Delete deletes the webhook app if it's deployed. Does nothing if it's not.
+// Returns true if the app was deleted.
+func Delete(ctx context.Context, cl runtimeclient.Client, s *runtime.Scheme, namespace string, oldObjectOnly bool) (bool, error) {
+	objs, err := getTemplateObjects(s, namespace, "0", []byte{00000001})
+	if err != nil {
+		return false, err
+	}
+
+	var deleted bool
+	for _, obj := range objs {
+		unst := &unstructured.Unstructured{}
+		unst.SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
+		objName := obj.GetName()
+		_, doNotDeleteFound := obj.GetLabels()["toolchain.dev.openshift.com/no-deletion"]
+		if doNotDeleteFound {
+			// this object needs to stay
+			continue
+		}
+		// TODO --- temporary migration step to delete the objects by using the old name
+		if oldObjectOnly {
+			oldName, found := obj.GetLabels()["toolchain.dev.openshift.com/old-name"]
```

Comment:
the key that is used in the manifests is `old-name` - without the `toolchain.dev.openshift.com/` prefix

Reply:
Great catch 🤦‍♂️ 

I've added the prefix in all annotations.